### PR TITLE
fix: handle invalid connections

### PR DESCRIPTION
**Pull request by markusahlstrand**
Did a bit of refactor of the zod schemas and added a try/catch for the connection validation

### CODE REVIEW

File: src/authentication-flows/social.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export async function socialAuth(
   });
 
   const oauthLoginUrl = new URL(connectionInstance.authorizationEndpoint);
-  if (authParams.scope) {
+  if (connectionInstance.scope) {
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:46]: I was passing the wrong scopes.. It's the scope of the connections and not whatever the client passed.

File: src/services/clients.ts [hunk 2]
```diff
@@ -12,16 +17,25 @@ export async function getClient(env: Env, clientId: string): Promise<Client> {
...
+      const defaultConnection =
+        defaultSettings?.connections?.find((c) => c.name === connection.name) ||
+        {};
 
-    return {
-      ...defaultConnection,
-      ...connection,
-    };
-  });
+      try {
```
Comment (markusahlstrand) [inline src/services/clients.ts:26]: Don't crash if the connection isn't valid. Just ignore

File: src/types/Client.ts [hunk 3]
```diff
@@ -4,74 +4,47 @@ import {
...
+  tokenEndpoint: z.string(),
+  responseType: z.custom<AuthorizationResponseType>().optional(),
+  responseMode: z.custom<AuthorizationResponseMode>().optional(),
+  createdAt: z.string(),
+  modifiedAt: z.string(),
 });
 
-// This can be decorated with default vaules
-export const PartialClientSchema = z.object({
+export const BaseClientSchema = z.object({
```
Comment (markusahlstrand) [inline src/types/Client.ts:42]: Create base schema that is extended. Less duplication

File: src/authentication-flows/social.ts (hunk 1 above)
Comment (danielgent) [reply src/authentication-flows/social.ts:46]: Easily done. I also reviewed this PR!

File: src/types/Client.ts [hunk 4]
```diff
@@ -90,5 +63,13 @@ export const PartialClientSchema = z.object({
   clientSecret: z.string(),
 });
 
+export const ClientSchema = BaseClientSchema.extend({
+  connections: z.array(ConnectionSchema),
+});
+
+export const PartialClientSchema = BaseClientSchema.extend({
```
Comment (danielgent) [inline src/types/Client.ts:70]: Nice. TIL

**Review by danielgent — APPROVED** [review]

**Pull request merged**